_posts/2021-03-11-Playing-along-with-NFTables.md

```diff
@@ -0,0 +1,294 @@
+table inet filter {
+    # chain names are up to you.
+    # what part of the traffic they cover, 
+    # depends on the type line.
+	chain input {
+		type filter hook input priority 0; policy accept;
+	}
+	chain forward {
+		type filter hook forward priority 0; policy accept;
+	}
+	chain output {
+		type filter hook output priority 0; policy accept;
+	}
+}
+```
+
+But so far we did not stop or allow any traffic. Well actually we let everything in and out now because all chains have the policy accept.
+
+```
+#!/usr/sbin/nft -f
+
+flush ruleset
+
+table inet filter {
+    chain base_checks {
+        ## another set, this time for connection tracking states.
+        # allow established/related connections
+        ct state {established, related} accept;
+
+        # early drop of invalid connections
+        ct state invalid drop;
+    }
+
+   	chain input {
+		type filter hook input priority 0; policy drop;
+        
+        # allow from loopback
+        iif "lo" accept;
+
+        jump base_checks;
+
+        # allow icmp and igmp
+        ip6 nexthdr icmpv6 icmpv6 type { echo-request, echo-reply, packet-too-big, time-exceeded, parameter-problem, destination-unreachable, packet-too-big, mld-listener-query, mld-listener-report, mld-listener-reduction, nd-router-solicit, nd-router-advert, nd-neighbor-solicit, nd-neighbor-advert, ind-neighbor-solicit, ind-neighbor-advert, mld2-listener-report } accept;
+        ip protocol icmp icmp type { echo-request, echo-reply, destination-unreachable, router-solicitation, router-advertisement, time-exceeded, parameter-problem } accept;
+        ip protocol igmp accept;
+        
+        # for testing reject with logging
+        counter log prefix "[nftables] input reject " reject;
+	}
+    chain forward {
+		type filter hook forward priority 0; policy accept;
+	}
+	chain output {
+		type filter hook output priority 0; policy accept;
+	}
+}
+```
+
+You can activate the configuration with `nft --file nftables.conf`, but do __NOT__ do this on a remote machine. It is also a good habit to run `nft --check --file nftables.conf` before actually loading the file to catch syntax errors.
+
+So what did we change?
+
+1. most importantly we changed the policy of the chain to drop and added a reject rule at the end. So nothing gets in right now.
+2. We allow all traffic on the localhost interface.
+3. The base_checks chain handles all packets related to established connections. This makes sure that incoming packets for outgoing connections get through.
+4. We allowed important ICMP/IGMP packets. Again this is using a set and the type names and not some crtyptic numbers. YAY for readability.
+
+Now if someome tries to do a ssh connect to our machine, we will see:
+
+```
+[Thu Mar 11 20:05:08 2021] [nftables] input reject IN=enp1s0 OUT= MAC=52:54:00:4c:51:6c:52:54:00:c7:a4:3a:08:00 SRC=172.16.16.11 DST=172.16.16.30 LEN=74 TOS=0x00 PREC=0x00 TTL=64 ID=54957 DF PROTO=UDP SPT=58099 DPT=53 LEN=54
```

Comment:
That's the wrong log line - ssh and DPT=53 don't mix too well ;-)

https://progress.opensuse.org/news/112 is already fixed, please take the log line (with DPT=22) from there.